pkg/logger/store.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2025 The KServe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logger
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/kserve/kserve/pkg/agent/storage"
+	"go.uber.org/zap"
+	"net/url"
+	"path"
+	"strings"
+)
+
+type StorageStrategy string
+
+const (
+	S3Storage   StorageStrategy = "s3"
+	HttpStorage StorageStrategy = "http"
+)
+
+const DefaultStorage = HttpStorage
+
+func GetStorageStrategy(url string) StorageStrategy {
+	// http, https
+	if strings.HasPrefix(url, "http") {
+		return HttpStorage
+	}
+
+	// s3, s3a
+	if strings.HasPrefix(url, "s3") {
+		return S3Storage
+	}
+
+	return DefaultStorage
```

Comment:
Better to have a switch case?
```suggestion
	switch {
	
	case strings.HasPrefix(url, "http"):  // http, https
		return HttpStorage
		
	case strings.HasPrefix(url, "s3"): // s3, s3a
		return S3Storage
		
	default:
		return DefaultStorage
	}
```